src/screens/home/swap-form/settings-button/settings-button.tsx

```diff
@@ -0,0 +1,105 @@
+import {useEffect, useState} from 'react';
+
+import styles from './settings-button.module.css';
+import {ToleranceButton} from './ToleranceButton/ToleranceButton';
+import {SettingsIcon} from '../../../../assets/icons/SettingsIcon/SettingsIcon';
+import {BottomSheet} from '../../../../components/bottom-sheet/bottom-sheet';
+import {FormButton} from '../../../../shared/FormButton/FormButton';
+import {useDispatch} from '../../../../store';
+import {setSlippageToleranceActions} from '../../../../store/settings/settings-actions';
+import {useSlippageToleranceSelector} from '../../../../store/settings/settings-selectors';
+
+export const SettingsButton = () => {
+    const dispatch = useDispatch();
+    const slippageTolerance = useSlippageToleranceSelector();
+    const [isOpen, setIsOpen] = useState(false);
+    const [slippageToleranceValue, setSlippageToleranceValue] =
+        useState(slippageTolerance);
```

Comment:
you can omit this state by using/setting `slippageTolerance` directly from state